riscv/src/register/mstatus.rs

```diff
@@ -72,6 +72,15 @@ impl From<bool> for Endianness {
 }
 
 impl Mstatus {
```

Comment:
It would be nice to have macros for structs like `Mstatus`, as I expect a lot of duplicates with this

Reply:
Maybe we could have a `pub(crate) bit` module somewhere with bitwise **functions** for `usize`. In this way, we avoid having methods in every register, and all registers use the same function. I'm not sure if this would help us optimize the binary size or if the compiler is smart enough to collapse all these methods (don't think so).

Reply:
Yeah I saw a lot of opportunity to collapse some repeated code.